### PR TITLE
add option to load pretrained weights for T5 model

### PR DESCRIPTION
**Pull request by prathikr**
### Description

Adds option to pass in pretrained weights file during T5 inference onnx export. Mimics the changes made to whisper: https://github.com/microsoft/onnxruntime/pull/15759

### Motivation and Context

Required for ONNX Runtime demo being presented at BUILD.

### CODE REVIEW

**Review by wangyems — DISMISSED** [review]

File: onnxruntime/python/tools/transformers/models/t5/convert_to_onnx.py [hunk 1]
```diff
@@ -147,10 +154,11 @@ def export_onnx_models(
     disable_auto_mixed_precision: bool = False,
     use_int32_inputs: bool = True,
     model_type: str = "t5",
+    state_dict_path: str = "",
```
Comment (baijumeswani) [inline onnxruntime/python/tools/transformers/models/t5/convert_to_onnx.py:157]: Shall we make it `None`?

```suggestion
    state_dict_path: Optional[str] = None,
```

File: onnxruntime/python/tools/transformers/models/t5/t5_helper.py [hunk 2]
```diff
@@ -66,6 +66,7 @@ def load_model(
         device: torch.device,
         merge_encoder_and_decoder_init: bool = True,
         model_type: str = "t5",
+        state_dict_path: str = "",
```
Comment (baijumeswani) [inline onnxruntime/python/tools/transformers/models/t5/t5_helper.py:69]:
```suggestion
        state_dict_path: Optional[str] = None,
```

File: onnxruntime/python/tools/transformers/models/t5/t5_helper.py [hunk 3]
```diff
@@ -84,6 +85,9 @@ def load_model(
             model = MT5ForConditionalGeneration.from_pretrained(model_name_or_path, cache_dir=cache_dir)
         else:
             raise ValueError("only support mode_type=t5 or mt5")
+        
+        if state_dict_path:
```
Comment (baijumeswani) [inline onnxruntime/python/tools/transformers/models/t5/t5_helper.py:89]:
```suggestion
        if state_dict_path is not None:
```

**Review by baijumeswani — APPROVED** [review]

File: onnxruntime/python/tools/transformers/models/t5/convert_to_onnx.py (hunk 1 above)
Comment (prathikr) [reply onnxruntime/python/tools/transformers/models/t5/convert_to_onnx.py:157]: Dismissing in favor of keeping the API consistent with whisper

File: onnxruntime/python/tools/transformers/models/t5/t5_helper.py (hunk 2 above)
Comment (prathikr) [reply onnxruntime/python/tools/transformers/models/t5/t5_helper.py:69]: Dismissing in favor of keeping the API consistent with whisper

**Review by baijumeswani — APPROVED** [review]

**Review by wangyems — APPROVED** [review]

**Pull request merged**